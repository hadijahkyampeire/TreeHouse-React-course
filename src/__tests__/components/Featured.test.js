import React from 'react';
import enzyme from 'enzyme';
import Featured from '../../components/Featured';

describe('About component', () => {
    const props = {
        match:{ params:{
                name: 'Hadijah',
                topic: 'Javascript'}  }
    }
    const node = enzyme.shallow(<Featured {...props} />);

    it('should render without crashing', () => {
        expect(node).toMatchSnapshot();
    });
});