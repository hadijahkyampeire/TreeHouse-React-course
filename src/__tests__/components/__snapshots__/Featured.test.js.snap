// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Featured
    match={
        Object {
            "params": Object {
              "name": "Hadijah",
              "topic": "Javascript",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Hadijah
           
</h2>,
        <p>
          Introducing 
          <strong>
                    Hadijah
          </strong>
          , a teacher who loves teaching courses about 
          <strong>
                    Javascript
          </strong>
          !
</p>,
      ],
      "className": "main-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hadijah",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Hadijah",
          " ",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Introducing ",
            <strong>
              Hadijah
</strong>,
            ", a teacher who loves teaching courses about ",
            <strong>
              Javascript
</strong>,
            "!",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Introducing ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hadijah",
            },
            "ref": null,
            "rendered": "Hadijah",
            "type": "strong",
          },
          ", a teacher who loves teaching courses about ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Javascript",
            },
            "ref": null,
            "rendered": "Javascript",
            "type": "strong",
          },
          "!",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Hadijah
             
</h2>,
          <p>
            Introducing 
            <strong>
                        Hadijah
            </strong>
            , a teacher who loves teaching courses about 
            <strong>
                        Javascript
            </strong>
            !
</p>,
        ],
        "className": "main-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hadijah",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Hadijah",
            " ",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Introducing ",
              <strong>
                Hadijah
</strong>,
              ", a teacher who loves teaching courses about ",
              <strong>
                Javascript
</strong>,
              "!",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Introducing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hadijah",
              },
              "ref": null,
              "rendered": "Hadijah",
              "type": "strong",
            },
            ", a teacher who loves teaching courses about ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Javascript",
              },
              "ref": null,
              "rendered": "Javascript",
              "type": "strong",
            },
            "!",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
